src/DIRAC/WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -345,12 +346,77 @@ def export_rescheduleJob(self, jobIDs):
     self.__sendJobsToOptimizationMind(validJobList)
     return result
 
-  def __deleteJob(self, jobID):
-    """ Delete one job
+  types_removeJob = []
 
-        :param int jobID: job ID
+  def export_removeJob(self, jobIDs):
+    """
+    Completely remove a list of jobs, also from TaskQueueDB,
+    and including its JobLogging info.
+    Only authorized users are allowed to remove jobs.
 
-        :return: S_OK()/S_ERROR()
+    :param list jobIDs: list of job IDs
+    :return: S_OK()/S_ERROR() -- confirmed job IDs
+    """
+
+    jobList = self.__getJobList(jobIDs)
+    if not jobList:
+      return S_ERROR('Invalid job specification: ' + str(jobIDs))
+
+    validJobList, invalidJobList, nonauthJobList, _ = self.jobPolicy.evaluateJobRights(jobList,
```

Comment:
is nonauthJobList a subset of invalidJobList? Otherwise the error reporting in line 400ff could be weird

Reply:
no I don't see `nonauthJobList` being a subset of `invalidJobList`